rustls/src/client/client_conn.rs

```diff
@@ -425,6 +437,30 @@ impl ClientConfig {
             .current_time()
             .ok_or(Error::FailedToGetCurrentTime)
     }
+
+    /// Reports whether we need to send the "key_share"
+    /// extension.
+    ///
+    /// This only returns false if we're using the PSK-only key
+    /// exchange mode.
+    pub(super) fn need_key_share(&self) -> bool {
```

Comment:
nit: may want to return an error if this is used with an incorrect version of TLS.
```suggestion
    pub(super) fn need_key_share(&self) -> Result<bool, Error> {
```

Reply:
Why? It's not an error. See the original code.